sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -116,7 +116,7 @@ private OperationInternal(OperationState<T> finalState)
             _stateLock = new AsyncLockWithValue<OperationState<T>>(finalState);
         }
 
-        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse;
+        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse ?? throw new InvalidOperationException("The operation has not completed yet.");
```

Comment:
```suggestion
        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : (_rawResponse ?? throw new InvalidOperationException("The operation has not completed yet."));
```

I'm not sure that this works in the current form and it may or may not need these parens.